Improve discoverability of internal endpoint namespacing (#1387)
ngrok/ngrok-docs: docs/universal-gateway/internal-endpoints.mdx

- End with the `.internal` domain extension
- Use the `internal` [binding](/docs/universal-gateway/bindings/)

:::note

> Internal domains are [namespaced](#namespacing) to the accounts that create them. Even if another user has the same internal endpoint URL as you, your traffic will never flow to their endpoint.

Reviewer: may as well enforce one sentence per line when making additions 😇

```suggestion
Internal domains are [namespaced](#namespacing) to the accounts that create them. 
Even if another user has the same internal endpoint URL as you, your traffic will never flow to their endpoint.
```

- **Scheme** - Internal endpoints support all [Endpoint Protocols](/universal-gateway/protocols) (`http`, `https`, `tcp` and `tls`).
- **Hostname** - Hostnames **must** end with `.internal` and may contain up to 5 subdomains. You must specify a URL with a hostname, randomly-assigned hostnames are not supported. You may specify wildcard hostnames.

> - **Port** - All port numbers [1-65535] are valid. For `tcp` endpoints, you **must** specify a port.

Reviewer: Maybe these should get their own sections too? (Feel freel to ignore if you'd rather not expand the scope beyond the immediate feedback)

> ## Namespacing

Reviewer: since this falls under `## URLs`

```suggestion
### Namespacing
```

Internal endpoints are namespaced on a per-account basis.

> This means that, two different accounts may have internal endpoints with the same URL (e.g. `https://api.internal`) without conflict or interference in [endpoint pooling](/universal-gateway/endpoint-pooling). As a result, there is no need to reserve an internal domain.

Reviewer: some gentle copyediting :)

```suggestion
This means that two different accounts may have internal endpoints with the same URL (such as `https://api.internal`) without conflict or interference in [endpoint pooling](/universal-gateway/endpoint-pooling). 
As a result, there is no need to reserve an internal domain.
```
